program-runtime/src/loaded_programs.rs

```diff
@@ -935,6 +956,26 @@ impl<FG: ForkGraph> LoadedPrograms<FG> {
         })
     }
 
+    /// Returns the list of loaded programs which are verified and compiled.
+    fn get_flattened_entries(&self) -> Vec<(Pubkey, Arc<LoadedProgram>)> {
```

Comment:
This could take two filter parameters `include_program_runtime_v1` and `include_program_runtime_v2`. Then `get_entries_sorted_by_tx_usage()`, which is only used by the recompilation phase could be inlined there.